src/platform/user/profile/utilities/index.js

```diff
@@ -123,18 +124,24 @@ export function mapRawUserDataToState(json) {
 export const hasSession = () => localStorage.getItem('hasSession');
 
 function compareLoginPolicy(loginPolicy) {
-  let attemptedLoginPolicy = localStorage.getItem(
-    authnSettings.PENDING_LOGIN_TYPE,
-  );
-
-  attemptedLoginPolicy =
-    attemptedLoginPolicy === 'mhv' ? 'myhealthevet' : attemptedLoginPolicy;
+  const attemptedLoginPolicy =
+    localStorage.getItem('pendingLoginPolicy') === 'mhv'
+      ? 'myhealthevet'
+      : localStorage.getItem('pendingLoginPolicy');
+
+  if (attemptedLoginPolicy === null) {
+    Raven.captureMessage(
+      "localStorage error: 'pendingLoginPolicy' was not stored",
+    );
+  }
 
   if (loginPolicy !== attemptedLoginPolicy) {
     recordEvent({
       event: `login-mismatch-${attemptedLoginPolicy}-${loginPolicy}`,
     });
   }
+
+  localStorage.removeItem('pendingLoginPolicy');
 }
 
 export function setupProfileSession(payload) {
```

Comment:
Is it possible this is getting called again after the localStorage keys have been used and discarded? As in after a successful login, the user is going back to the auth callback page for some reason. Might be worth having the function quick return if there's already an active session.

```
if (hasSession()) return;
```

Reply:
Or perhaps at least after setting the userFirstName. Anything after that, we should probably not report extra analytics.

Reply:
Although, my rationale for including setting the name is that it could be a new active user somehow. But then we would have failed to report the GA event for the LOA for the user if it's a new user. Perhaps that event should move up so that all of the reporting around login type can be in a contiguous block. At which point it might make sense to move all of that into its own function.

Reply:
@U-DON I think your theory about it getting called again after the keys have been discarded is a good one.  I was a little unclear about your firstName comment, but I moved the GA event recording into its own function, placed it earlier in the block, and am only calling it if `hasSession() === false`.  Take a look and let me know if it covers what you were mentioning.